### DISCUSSION TITLE
Leaning camera roll

### DISCUSSION THREAD

**Discussion by AdamTadeusz** [NeotokyoRebuild/neo#647] [Ideas]
See https://github.com/NeotokyoRebuild/neo/pull/638 for more background, but essentially the current default leaning values of body_yaw_left_amount 7.0 (should be 7.5) and body_yaw_right_amount 15.0 result in exactly the same amount of a persons body being exposed around a corner, and therefore the same amount of information gained from peeking around said corner, as in the original.

The one caveat is that currently in the original game the camera roll is independent of the body roll, so regardless of which way you lean and how much your character is leaning, your camera will be rotated 20 degrees in the direction in which you are leaning. While this might have the placebo effect of making the lean seem stronger than it actually is, it also has the undesired effect of making controlling the cursor more difficult as the screen is rotated 20 degrees.

I propose we keep the current functionality of NT;RE, where the camera roll depends on the body roll so a left peek would result in 7.5 degrees of camera roll and a right peek would result in 15 degrees of camera roll. We could go one step further as people tend to naturally lean their head away from the direction in which they are leaning to maintain a level head, and instead just rotate the view model, but that might be too big a departure from the original game (games like double action boogaloo and the upcoming out of action game do this when doing a full roll to prevent motion sickness and maintain awareness)

**Comment by AdamTadeusz** [comment]
https://github.com/user-attachments/assets/bea49adc-eca5-456d-bf28-c1dba566123a

Here's what a lean with no camera roll but with view model roll would look like, in reference to that last suggestion.

**Comment by Agiel** [comment]
I don't see a good reason to mimic the full 20 degree roll as OG NT, so I'd say let's go with having the view roll match the body roll as you suggested.

There's probably no harm in letting players disable roll as an option, but I don't think it should be the default, as roll gives you a better sense of where your body is in space and lets you better judge how much you're exposing.
